pkg/api/api_implementation.go

```diff
@@ -439,3 +440,23 @@ func addPaginationQueryParameters(req *http.Request, opts types.ListOptions) {
 
 	req.URL.RawQuery = query.Encode()
 }
+
+func decodeResponse(ctx context.Context, mediaType string, data io.Reader, res interface{}) error {
+	if mediaType == "application/json" {
+		if decodeResponse, ok := res.(types.ResponseDecodeHook); ok {
```

Comment:
one of the iterations included a check for `if not a List operation`. We don't actually need it, since List operations decode into a `[]json.RawMessage`, which never is a `types.ResponseDecodeHook`

Reply:
Good catch...
I'll just pretend I did this on purpose, since my life is of course always guided by PEP20.

On a more serious note, maybe some documentation how these routines interact with the different methods might be helpful for future development.

Reply:
"Flat is better than nested."? :stuck_out_tongue: 

More docs would definitely be helpful. Do you have any specific suggestions what to document? Having a hard time finding _what_ to document .. writing the docs isn't that big of a problem.